tsfc/fem.py

```diff
@@ -233,6 +234,11 @@ def flat_index(ordered_deriv):
         return result[0]
 
 
+def zero_tracking_literal(table, ufl_element):
+    track_zeros = isinstance(ufl_element, (ufl.EnrichedElement, ufl.MixedElement))
```

Comment:
So I did this to exactly mimic the old FFC behaviour.
@FabioLuporini: Is there any reason not to track zeros for all matrices? Code would be much simpler and with better separation of concerns.